compiler/src/test/java/com/squareup/anvil/compiler/codegen/BindingModuleGeneratorTest.kt

```diff
@@ -173,6 +173,102 @@ class BindingModuleGeneratorTest(
     }
   }
 
+  @Test fun `the bound type is implied when there is only one super type`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        import com.squareup.anvil.annotations.ContributesBinding
+        $import
+
+        interface ParentInterface
+
+        @ContributesBinding(Any::class)
+        interface ContributingInterface : ParentInterface
+        
+        $annotation(Any::class)
+        interface ComponentInterface
+    """
+    ) {
+      val modules = if (annotationClass == MergeModules::class) {
+        componentInterface.daggerModule.includes.toList()
+      } else {
+        componentInterface.anyDaggerComponent.modules
+      }
+      assertThat(modules).containsExactly(componentInterfaceAnvilModule.kotlin)
+
+      val methods = modules.first().java.declaredMethods
+      assertThat(methods).hasLength(1)
+
+      with(methods[0]) {
+        assertThat(returnType).isEqualTo(parentInterface)
+        assertThat(parameterTypes.toList()).containsExactly(contributingInterface)
+        assertThat(isAbstract(modifiers)).isTrue()
+        assertThat(isAnnotationPresent(Binds::class.java)).isTrue()
+      }
+    }
+  }
+
+  @Test fun `the bound type can only be implied with on super type (2 interfaces)`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        import com.squareup.anvil.annotations.ContributesBinding
+        $import
+
+        interface ParentInterface
+        
+        @ContributesBinding(Any::class)
+        interface ContributingInterface : ParentInterface, CharSequence
+        
+        $annotation(Any::class)
+        interface ComponentInterface
+    """
+    ) {
+      assertThat(exitCode).isEqualTo(COMPILATION_ERROR)
+
+      assertThat(messages).contains("Source.kt: (9, 11)")
+      assertThat(messages).contains(
+          "com.squareup.test.ContributingInterface contributes a binding, but does not specify " +
+              "the bound type. This is only allowed with one direct super type. If there are " +
+              "multiple, then the bound type must be explicitly defined in the " +
+              "@ContributesBinding annotation."
+      )
+    }
+  }
+
+  @Test fun `the bound type can only be implied with on super type (class and interface)`() {
+    compile(
+        """
+        package com.squareup.test
+        
+        import com.squareup.anvil.annotations.ContributesBinding
+        $import
+
+        interface ParentInterface
+        
+        open class Abc
+        
+        @ContributesBinding(Any::class)
+        interface ContributingInterface : Abc(), ParentInterface
+        
+        $annotation(Any::class)
+        interface ComponentInterface
+    """
+    ) {
+      assertThat(exitCode).isEqualTo(COMPILATION_ERROR)
+
+      assertThat(messages).contains("Source.kt: (11, 11)")
+      assertThat(messages).contains(
+          "com.squareup.test.ContributingInterface contributes a binding, but does not specify " +
```

Comment:
I may have mentioned this before, but all these tests are going to fail on you (and be annoying) if you end up refactoring the wording of your logging strings. I wonder if you could leverage something Android-esque like strings.xml to have a centralized spot for them.

Not a ship stopper just something to think about moving forward.

Reply:
In older tests I didn't check the error message at all and only the position. But I found those tests actually harder to maintain and read. A centralized spot could help, but is also hard to work with due to the placeholders. For now I think that's the best option knowing that I will change my mind in a few days :-)